### PR TITLE
Fixes several bugs in the lifecycle management of RosActivities and adds DiagnosticsArrayView.

### PR DESCRIPTION
**Pull request by damonkohler**

### CODE REVIEW

**Comment by damonkohler** [comment]
PTAL @chadrockey and @stonier.

File: android_10/src/org/ros/android/NodeMainExecutorService.java [hunk 1]
```diff
@@ -127,20 +135,31 @@ public void shutdownNodeMain(NodeMain nodeMain) {
...
+      public void run() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(NodeMainExecutorService.this);
+        builder.setMessage("Continue shutting down?");
+        builder.setPositiveButton("Shutdown", new OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialog, int which) {
+            forceShutdown();
+          }
+        });
+        builder.setNegativeButton("Cancel", new OnClickListener() {
```
Comment (chadrockey) [inline android_10/src/org/ros/android/NodeMainExecutorService.java:149]: This pops up a dialog to confirm whether to shut down the service?  I think we should have a clearer label than "Shutdown" or "Cancel".

Maybe "Shutdown" or "Nevermind"
       or "Shutdown" or "Keep Running"

**Comment by chadrockey** [comment]
The DiagnosticArrayView looks good to me.

The other commits look like real improvements to me.  I hate accidentally shutting down apps.  Any chance this helps prevent the background service from being disconnected and then duplicated?

File: android_10/src/org/ros/android/NodeMainExecutorService.java (hunk 1 above)
Comment (damonkohler) [inline android_10/src/org/ros/android/NodeMainExecutorService.java:149]: "Keep Running" isn't bad. But I think users expect a "Cancel" button. Plus when you read the "Continue shutting down?" question, it seems to make sense. I'll run it by a UXer here and go with his decision. SG?

**Comment by damonkohler** [comment]
I'm not sure I've seen the duplication problem you're having. I do see a ROS starting up after shutdown though. I'm still looking into that.

File: android_10/src/org/ros/android/NodeMainExecutorService.java (hunk 1 above)
Comment (SirVer) [inline android_10/src/org/ros/android/NodeMainExecutorService.java:149]: I agree with chad. "Keep Running" is clear enough for a cancel button. I think a lot of android dialogs rely on the user tapping outside of it to cancel, so I would not say that the user expects a cancel button.

File: android_10/src/org/ros/android/NodeMainExecutorService.java [hunk 2]
```diff
@@ -203,7 +232,7 @@ public void setMasterUri(URI uri) {
 
   /**
    * This version of startMaster can only create private masters.
-   *
+   *
```
Comment (SirVer) [inline android_10/src/org/ros/android/NodeMainExecutorService.java]: whitespace.

Comment (damonkohler) [inline android_10/src/org/ros/android/NodeMainExecutorService.java]: Done.

File: android_10/src/org/ros/android/NodeMainExecutorService.java (hunk 1 above)
Comment (damonkohler) [inline android_10/src/org/ros/android/NodeMainExecutorService.java:149]: UX says "Cancel" is the way to go.

**Pull request merged**